src/pairing.cpp

```diff
@@ -351,19 +354,42 @@ void pairing(vector<unique_ptr<RoughReservoir>> &upper_reservoirs,
           upper_reservoir->pour_point, lower_reservoir->pour_point, coslat);
 
       if(upper_reservoir->brownfield){
+        distance_sqd = INF;
         RoughBfieldReservoir* br = static_cast<RoughBfieldReservoir*>(upper_reservoir);
-        for(ArrayCoordinate ac: br->shape_bound)
-          distance_sqd =
-              MIN(find_distance_sqd(ac, lower_reservoir->pour_point, coslat), distance_sqd);
+        for(size_t i = 0; i<br->shape_bound.size(); i++){
+          ArrayCoordinate ac = br->shape_bound[i];
+          double dist_sqd = find_distance_sqd(ac, lower_reservoir->pour_point, coslat);
+          if(dist_sqd < distance_sqd){
+            distance_sqd = dist_sqd;
+          }
+        }
       }
       if(lower_reservoir->brownfield || lower_reservoir->ocean){
+        distance_sqd = INF;
```

Comment:
Same here, perhaps min_distance_sqd for clarity

Reply:
Fixed